In [1]:
from pathlib import Path
import sys
import pandas as pd

PROJECT_ROOT = Path.cwd().parent
if str(PROJECT_ROOT) not in sys.path:
    sys.path.append(str(PROJECT_ROOT))

from src.config import EXPERIMENT_LOG_FILE

log_df = pd.read_csv(EXPERIMENT_LOG_FILE)

metric_cols = ["test_MAE", "test_RMSE", "test_MAPE", "test_sMAPE", "test_R2"]
for col in metric_cols:
    log_df[col] = pd.to_numeric(log_df[col], errors="coerce")

log_df.head()

,run_id,timestamp,target,dataset_name,frequency,detected_frequency,feature_set_name,data_mode,model_name,model_params,...,test_MAE,test_RMSE,test_MAPE,test_sMAPE,test_R2,model_path,prediction_path,diagnostic_plot_path,feature_importance_path,params_path
0,run_20260326_135954_1beb8be6,2026-03-26 13:59:54,BA_TotActPwr_BESS_AC_Panel1,Imputed_data,5min,5min,all_features_v1,dense,linear_regression,"{""copy_X"": true, ""fit_intercept"": true, ""n_job...",...,0.227493,0.593984,24.804799,15.843181,0.927327,C:\Data_analysis\Thesis\outputs\models\run_202...,C:\Data_analysis\Thesis\outputs\predictions\ru...,C:\Data_analysis\Thesis\outputs\figures\foreca...,C:\Data_analysis\Thesis\outputs\feature_import...,C:\Data_analysis\Thesis\outputs\logs\run_20260...
1,run_20260326_135955_2ded4f36,2026-03-26 13:59:55,BA_TotActPwr_BESS_AC_Panel1,Imputed_data,5min,5min,all_features_v1,dense,ridge,"{""alpha"": 1.0, ""copy_X"": true, ""fit_intercept""...",...,0.227476,0.593980,24.802236,15.840235,0.927328,C:\Data_analysis\Thesis\outputs\models\run_202...,C:\Data_analysis\Thesis\outputs\predictions\ru...,C:\Data_analysis\Thesis\outputs\figures\foreca...,C:\Data_analysis\Thesis\outputs\feature_import...,C:\Data_analysis\Thesis\outputs\logs\run_20260...
2,run_20260326_135956_f45e2aad,2026-03-26 13:59:56,BA_TotActPwr_BESS_AC_Panel1,Imputed_data,5min,5min,all_features_v1,dense,random_forest,"{""bootstrap"": true, ""ccp_alpha"": 0.0, ""criteri...",...,0.274168,0.675291,29.245611,16.788629,0.906070,C:\Data_analysis\Thesis\outputs\models\run_202...,C:\Data_analysis\Thesis\outputs\predictions\ru...,C:\Data_analysis\Thesis\outputs\figures\foreca...,C:\Data_analysis\Thesis\outputs\feature_import...,C:\Data_analysis\Thesis\outputs\logs\run_20260...
3,run_20260326_140213_03187aa3,2026-03-26 14:02:13,BA_TotActPwr_BESS_AC_Panel1,Imputed_data,5min,5min,all_features_v1,dense,xgboost,"{""objective"": ""reg:squarederror"", ""base_score""...",...,0.243553,0.727950,20.211308,11.535189,0.890849,C:\Data_analysis\Thesis\outputs\models\run_202...,C:\Data_analysis\Thesis\outputs\predictions\ru...,C:\Data_analysis\Thesis\outputs\figures\foreca...,C:\Data_analysis\Thesis\outputs\feature_import...,C:\Data_analysis\Thesis\outputs\logs\run_20260...
4,run_20260326_140231_10cb28a9,2026-03-26 14:02:31,BA_TotActPwr_BESS_AC_Panel1,Data_with_NaNs,5min,5min,all_features_v1,nan_friendly,hist_gbr,"{""categorical_features"": ""from_dtype"", ""early_...",...,0.226474,0.695078,21.175391,12.846201,0.900681,C:\Data_analysis\Thesis\outputs\models\run_202...,C:\Data_analysis\Thesis\outputs\predictions\ru...,C:\Data_analysis\Thesis\outputs\figures\foreca...,NaN,C:\Data_analysis\Thesis\outputs\logs\run_20260...


In [2]:
academy_results = log_df[
    (log_df["target"] == "BU_TotActPwr_Tech_Room") &
    (log_df["feature_set_name"] == "all_features_v1")
].copy()  #    (log_df["target"] == "BU_TotActPwr_Tech_Room") &

academy_results = academy_results.sort_values("test_RMSE", ascending=True)

academy_results[
    [
        "run_id",
        "model_name",
        "target",
        "dataset_name",
        "data_mode",
        "frequency",
        "feature_set_name",
        "test_MAE",
        "test_RMSE",
        "test_MAPE",
        "test_sMAPE",
        "test_R2",
    ]
]

,run_id,model_name,target,dataset_name,data_mode,frequency,feature_set_name,test_MAE,test_RMSE,test_MAPE,test_sMAPE,test_R2
47,run_20260326_154931_8ed0b63f,xgboost,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,15min,all_features_v1,0.085095,0.162799,5.577106,5.274815,0.987085
46,run_20260326_154926_46b36d2e,hist_gbr,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,15min,all_features_v1,0.087242,0.169788,5.666668,5.340051,0.985953
45,run_20260326_154922_31a527a4,xgboost,BU_TotActPwr_Tech_Room,Imputed_data,dense,15min,all_features_v1,0.090972,0.172347,6.118888,5.658857,0.985611
44,run_20260326_154859_42489288,random_forest,BU_TotActPwr_Tech_Room,Imputed_data,dense,15min,all_features_v1,0.091427,0.180622,5.675108,5.322888,0.984196
48,run_20260326_154936_5b452af6,catboost,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,15min,all_features_v1,0.098916,0.181105,6.133532,5.839951,0.984018
52,run_20260326_155207_f9a2d458,xgboost,BU_TotActPwr_Tech_Room,Imputed_data,dense,5min,all_features_v1,0.094804,0.192072,6.081118,5.539115,0.982288
54,run_20260326_155218_608181c3,xgboost,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,5min,all_features_v1,0.097528,0.192141,6.035875,5.552447,0.982253
53,run_20260326_155212_e3f6694c,hist_gbr,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,5min,all_features_v1,0.099742,0.197695,6.032272,5.602039,0.981212
51,run_20260326_155037_96cfe428,random_forest,BU_TotActPwr_Tech_Room,Imputed_data,dense,5min,all_features_v1,0.094316,0.198051,6.105730,5.558563,0.981169
55,run_20260326_155226_e0c9bc90,catboost,BU_TotActPwr_Tech_Room,Data_with_NaNs,nan_friendly,5min,all_features_v1,0.104481,0.203975,6.478658,5.962691,0.979999


In [3]:
academy_compare = log_df[
    (log_df["feature_set_name"] == "all_features_v1")
].copy()   #(log_df["target"] == "BU_TotActPwr_Academy") &

academy_compare["test_RMSE"] = pd.to_numeric(academy_compare["test_RMSE"], errors="coerce")
academy_compare = academy_compare.dropna(subset=["test_RMSE"])

best_per_dataset = (
    academy_compare.sort_values("test_RMSE", ascending=True)
    .groupby("dataset_name", as_index=False)
    .first()
)

best_per_dataset[
    [
        "dataset_name",
        "model_name",
        "target",
        "data_mode",
        "test_MAE",
        "test_RMSE",
        "test_MAPE",
        "test_sMAPE",
        "test_R2",
    ]
]

,dataset_name,model_name,target,data_mode,test_MAE,test_RMSE,test_MAPE,test_sMAPE,test_R2
0,Data_with_NaNs,xgboost,BU_TotActPwr_Tech_Room,nan_friendly,0.085095,0.162799,5.577106,5.274815,0.987085
1,Imputed_data,xgboost,BU_TotActPwr_Tech_Room,dense,0.090972,0.172347,6.118888,5.658857,0.985611


In [4]:
academy_compare = log_df[

    (log_df["feature_set_name"] == "all_features_v1")
].copy()
#  (log_df["target"] == "BU_TotActPwr_Academy") &
academy_compare = academy_compare.sort_values(
    ["dataset_name", "test_RMSE"], ascending=[True, True]
)

academy_compare[
    [
        "run_id",
        "model_name",
        "dataset_name",
        "target",
        "data_mode",
        "test_MAE",
        "test_RMSE",
        "test_MAPE",
        "test_sMAPE",
        "test_R2",
    ]
]

,run_id,model_name,dataset_name,target,data_mode,test_MAE,test_RMSE,test_MAPE,test_sMAPE,test_R2
47,run_20260326_154931_8ed0b63f,xgboost,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.085095,0.162799,5.577106,5.274815,0.987085
46,run_20260326_154926_46b36d2e,hist_gbr,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.087242,0.169788,5.666668,5.340051,0.985953
48,run_20260326_154936_5b452af6,catboost,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.098916,0.181105,6.133532,5.839951,0.984018
54,run_20260326_155218_608181c3,xgboost,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.097528,0.192141,6.035875,5.552447,0.982253
53,run_20260326_155212_e3f6694c,hist_gbr,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.099742,0.197695,6.032272,5.602039,0.981212
55,run_20260326_155226_e0c9bc90,catboost,Data_with_NaNs,BU_TotActPwr_Tech_Room,nan_friendly,0.104481,0.203975,6.478658,5.962691,0.979999
58,run_20260327_085035_b375ea4c,catboost,Data_with_NaNs,BU_TotActPwr_SDB_EL_Substation,nan_friendly,0.345614,0.488030,5.291563,5.287460,0.867431
56,run_20260327_085019_0009dc76,hist_gbr,Data_with_NaNs,BU_TotActPwr_SDB_EL_Substation,nan_friendly,0.367030,0.509970,5.811188,5.776749,0.855243
57,run_20260327_085028_3d3b1bad,xgboost,Data_with_NaNs,BU_TotActPwr_SDB_EL_Substation,nan_friendly,0.444999,0.575930,6.734293,6.800945,0.815375
4,run_20260326_140231_10cb28a9,hist_gbr,Data_with_NaNs,BA_TotActPwr_BESS_AC_Panel1,nan_friendly,0.226474,0.695078,21.175391,12.846201,0.900681


# Best Model Summary

| Target | Best Model Summary |
|---|---|
| **BU_TotActPwr_Tech_Room** | **NaNs:** XGBoost (RMSE **0.1628**)  •  **Imputed:** XGBoost (RMSE **0.1723**)  → **NaNs slightly better** |
| **BU_TotActPwr_SDB_EL_Substation** | **NaNs:** CatBoost (RMSE **0.4880**)  •  **Imputed:** Not available |
| **BA_TotActPwr_BESS_AC_Panel1** | **NaNs:** HistGBR (RMSE **0.6951**)  •  **Imputed:** Linear/Ridge (RMSE **0.5940**)  → **Imputed better** |
| **BA_TotActPwr_BESS_AC_Panel2** | **NaNs:** HistGBR (RMSE **0.7042**)  •  **Imputed:** Linear/Ridge (RMSE **0.6439**)  → **Imputed better** |
| **BU_TotActPwr_Academy** | **NaNs:** HistGBR (RMSE **1.9755**)  •  **Imputed:** Linear/Ridge (RMSE **1.5579**)  → **Imputed better** |

## Overall
In most targets, **Imputed Data performed better**.  
The main exception is **Tech Room**, where **Data with NaNs performed slightly better**.